utam-compiler/src/main/java/utam/compiler/translator/ClassSerializer.java

```diff
@@ -95,6 +97,15 @@ public String toString() {
             .filter(method -> method.isPublic() || translationContext.getUsedPrivateMethods().contains(method.getDeclaration().getName()))
             .flatMap(method -> getMethodDeclaration(method).stream()).forEach(out::add);
     out.add(NEW_LINE);
+    source.getMethods().stream()
+        .filter(method -> (method.isPublic() || translationContext.getUsedPrivateMethods().contains(method.getDeclaration().getName())) && method.isElementMethod())
+        .map(method -> method.getDeclaration().getReturnType())
+        .map(returnType -> returnType instanceof TypeUtilities.ListOf ? ((TypeUtilities.ListOf)returnType).getElementType() : returnType)
+        .map(returnType -> String.format(
+            "public static class %s extends BasePageElement implements %s {}",
```

Comment:
- i'd prefer to use BasePageElement.class instead hardcoding String. less brittle If we rename base class
- why class has to be public? i understand interface, but class?

Reply:
Good catch on `BasePageElement`. Using `.class.getSimpleName()` in next commit.

I'm sure I just defaulted it to `public`. No other reason. I'll see about making it package private.